BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -291,11 +300,32 @@ extension StackLayout {
             space = minimumSpacing
         case .spaceEvenly:
             space = (layoutSize.axis - totalBasisSize) / CGFloat(basisSizes.count-1)
+        case .justifyStart:
+            space = minimumSpacing
+        case .justifyMiddle:
+            space = minimumSpacing
+        case .justifyEnd:
+            space = minimumSpacing
         }
 
         var frames = _calculateCross(basisSizes: basisSizes, layoutSize: layoutSize)
 
-        var axisOrigin: CGFloat = 0.0
+        var axisOrigin: CGFloat
+
+        switch underflow {
+        case .growProportionally:
+            axisOrigin = 0.0
+        case .growUniformly:
+            axisOrigin = 0.0
+        case .spaceEvenly:
+            axisOrigin = 0.0
+        case .justifyStart:
+            axisOrigin = 0.0
+        case .justifyMiddle:
+            axisOrigin = extraSize / 2.0
```

Comment:
We'll want to round this to the nearest point. (You could also add a TODO marker to remind me to come add screen scale rounding to this after #18 lands, if you want.) And then probably also want to add a test that centers an odd-sized child in an even-sized stack (or vice versa).